sdk/rust/README.md

```diff
@@ -2,80 +2,107 @@
 
 The Spin Rust SDK makes it easy to build Spin components in Rust.
 
-### Writing Spin HTTP components in Rust
+### Writing Spin HTTP Components in Rust
 
 This library simplifies writing Spin HTTP components. Below is an example of
 such a component:
 
 ```rust
 // lib.rs
-use anyhow::Result;
-use spin_sdk::{
-    http::{Request, Response},
-    http_component,
-};
+use spin_sdk::http::{IntoResponse, Request, Response};
+use spin_sdk::http_component;
 
 /// A simple Spin HTTP component.
 #[http_component]
-fn hello_world(req: Request) -> Result<Response> {
-    println!("{:?}", req.headers);
-    Ok(Response::new_with_headers(200, &[] "Hello, Fermyon!"))
+fn handle_hello_world(req: Request) -> anyhow::Result<impl IntoResponse> {
+    println!("Handling request to {:?}", req.header("spin-full-url"));
+    Ok(Response::builder()
+        .status(200)
+        .header("content-type", "text/plain")
+        .body("Hello, Fermyon")
+        .build())
 }
 ```
 
-The important things to note in the function above:
+The important things to note about the function above are:
 
-- the `spin_sdk::http_component` macro — this marks the function as the
-  entrypoint for the Spin component
-- the function signature — `fn hello_world(req: Request) -> Result<Response>` —
-`req` can be any number of types including the built in `Request` type or 
-the `http::Request` from the popular `http` crate. Likewise, the response type
-can be many things including the built in `Response` type or the `http::Response` type
-from the `http` crate.
+- the `spin_sdk::http_component` macro marks the function as the entry point for the Spin component,
+- in the function signature (`fn handle_hello_world(req: Request) -> anyhow::Result<impl IntoResponse>`), `req` can be any number of types, including the built-in `Request` type or the `http::Request` from the popular `http` crate
+- in the function signature, the response type can be many things, including `anyhow::Result<impl IntoResponse>` (as shown above) or the `http::Response` type from the `http` crate (e.g. `Result<Response>`). Note - Using the `http` crate will require you to add it, e.g. `cargo install http`.
 
-### Making outbound HTTP requests
+### Making Outbound HTTP Requests
 
-Let's see an example where the component makes an outbound HTTP request to a
-server, modifies the result, then returns it:
+Let's see an example where the component makes an outbound HTTP request to a server, modifies the result, and then returns it:
 
 ```rust
+use anyhow::Result;
```

Comment:
We don't import `anyhow::Result` in the example above so I would remove this and change line 45 to:

```rust
async fn handle_hello_world(_req: Request) -> anyhow::Result<impl IntoResponse> {
```

Reply:
Done!